### PR TITLE
Bugfix/enable master key for selected ips

### PR DESCRIPTION
**Pull request by AlexisNeveu**
Enables usage of masterKey by adding the IPs allowed to use it.

### CODE REVIEW

File: src/models/Configuration.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ class Configuration {
    * @param {string} [props.parseServer.masterKey] - A special access key that grants
    * unrestricted access to the Parse backend, allowing administrative-level operations.
    * and data management.
+   * @param {Array.<string>} [props.parseServer.masterKeyIps] - Restrict masterKey to be used
```
Comment (Zorin95670) [inline src/models/Configuration.js:21]: - string[]
- Missing `.`
- `Restricted ip that can use masterKey.`

File: src/models/Configuration.js [hunk 2]
```diff
@@ -70,6 +73,11 @@ class Configuration {
      * @type {string}
      */
     this.parseServer.masterKey = props.parseServer?.masterKey || null;
+    /**
+     * Restrict masterKey to be used by only these ips
+     * @type {Array.<string>}
```
Comment (Zorin95670) [inline src/models/Configuration.js:78]: missing default

File: src/services/ConfigurationService.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ export function loadConfiguration() {
       databaseURI: process.env.DATABASE_URI || '', // ,
       appId: process.env.APP_ID || 'leto-modelizer-api',
       masterKey: process.env.MASTER_KEY || '',
+      masterKeyIps: process.env.MASTER_KEY_IPS?.split(' ') || ['0.0.0.0/0'],
```
Comment (Zorin95670) [inline src/services/ConfigurationService.js:19]: `process.env.MASTER_KEY_IPS?.split(','),`

File: README.md [hunk 4]
```diff
@@ -75,6 +75,7 @@ Now your project is up and running using Docker, providing a convenient and isol
 | DATABASE_URI | Yes                                      | postgres://leto_admin:password@localhost:5432/leto_db | Represents a connection string used to connect to a database.                                                                            |
 | APP_ID       | No, default: `leto-modelizer-api`        | leto-modelizer-api                                    | A unique identifier for your application, used to distinguish it when interacting with the Parse backend.                                |
 | MASTER_KEY   | Yes                                      | mySuperSecretMasterKey123                             | A special access key that grants unrestricted access to the Parse backend, allowing administrative-level operations and data management. |
+| MASTER_KEY_IPS | No, default: 0.0.0.0/0               | 0.0.0.0/0                                             | Restrict masterKey to be used by only these ips. 0.0.0.0/0 enables any ip |
```
Comment (Zorin95670) [inline README.md:78]: Bad indentation

File: README.md [hunk 5]
```diff
@@ -89,6 +90,7 @@ configuration settings. Add the following variables to the .env file:
 DATABASE_URI=postgres://leto_admin:password@localhost:5432/leto_db
 APP_ID=my_app_id
 MASTER_KEY=my_super_secret_master_key
+MASTER_KEY_IPS=1.1.1.1/1 2.2.2.2/2 3.3.3.3/3
```
Comment (Zorin95670) [inline README.md:93]: add `,` instead of ` `

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -70,15 +70,16 @@ Now your project is up and running using Docker, providing a convenient and isol
...
+| DATABASE_URI   | Yes                                      | postgres://leto_admin:password@localhost:5432/leto_db | Represents a connection string used to connect to a database.                                                                            |
+| APP_ID         | No, default: `leto-modelizer-api`        | leto-modelizer-api                                    | A unique identifier for your application, used to distinguish it when interacting with the Parse backend.                                |
+| MASTER_KEY     | Yes                                      | mySuperSecretMasterKey123                             | A special access key that grants unrestricted access to the Parse backend, allowing administrative-level operations and data management. |
+| MASTER_KEY_IPS | No, default: [], no ip allowed           | 0.0.0.0/0                                             | Restrict masterKey to be used by only these ips. 0.0.0.0/0 enables any ip, not recommended for production                                |
```
Comment (aboubakar14) [inline README.md:78]: Missing dot at the end of the sentence in the description

**Review by aboubakar14 — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**